sdks/router-sdk/src/entities/trade.test.ts

```diff
@@ -627,6 +709,78 @@ describe('Trade', () => {
       expect(trade.tradeType).toEqual(TradeType.EXACT_INPUT)
     })
 
+    it('can be constructed with ETHER as input for exact input swap, with V4 eth route and V2 weth route', async () => {
+      const routeOriginalV2 = new V2RouteSDK([pair_weth_0, pair_0_1], ETHER, token1)
+      const routev2 = new RouteV2(routeOriginalV2)
+      const amountv2 = CurrencyAmount.fromRawAmount(ETHER, JSBI.BigInt(100))
+
+      const routeOriginalV4 = new V4RouteSDK([pool_v4_1_eth], ETHER, token1)
+      const routev4 = new RouteV4(routeOriginalV4)
+      const amountv4 = CurrencyAmount.fromRawAmount(ETHER, JSBI.BigInt(1000))
+
+      const trade = await Trade.fromRoutes(
+        [{ routev2, amount: amountv2 }],
+        [],
+        TradeType.EXACT_INPUT,
+        [],
+        [{ routev4, amount: amountv4 }]
+      )
+
+      expect(trade.tradeType).toEqual(TradeType.EXACT_INPUT)
+      expect(trade.inputAmount.currency).toEqual(ETHER)
+      expect(trade.outputAmount.currency).toEqual(token1)
+      expect(trade.swaps.length).toEqual(2)
+      expect(trade.routes.length).toEqual(2)
+      // Expect all swap input amounts to be native
+      expect(trade.swaps.every((swap) => swap.inputAmount.currency.isNative)).toBe(true)
+      // Expect all route inputs to be ETH
+      expect(trade.swaps.every((swap) => swap.route.input.isNative)).toBe(true)
+      // However, expect the routes to be preserved (v2 using WETH and v4 using ETH)
+      expect(trade.swaps[0].route.pathInput).toEqual(weth)
+      expect(trade.swaps[1].route.pathInput).toEqual(ETHER)
+
+      // Expect inputAmountNative to correctly track only the amount required for the ETH input V4 route
+      expect(trade.amounts.inputAmountNative).toBeDefined()
+      expect(trade.amounts.inputAmountNative?.equalTo(amountv4)).toBe(true)
+      expect(trade.amounts.outputAmountNative).toBeUndefined()
```

Comment:
I notice that the new `amounts.inputAmount` and `amounts.outputAmount` are never tested
i guess i would additionally check in 1 test that they have the expected amounts too to be 100% sure

i.e. `trade.amounts.inputAmount` equals `v4Amount + v2Amount`? and I actually dont know what `outputAmount` would equal on an exactIn test is it the worst case slippage amount?

Reply:
I added tests for inputAmount and outputAmount for these two test cases, and yeah inputAmount is the sum, outputAmount is hard to test for exact input like you said but its the same calculations as the original getter `trade.outputAmount`

Reply:
Also added two more test cases